### PR TITLE
[onert] Pass execution option into Execution and Executor

### PR DESCRIPTION
**Pull request by hseok-oh**
This commit updates Execution and Executor interface to get execution option from API and use.

ONE-DCO-1.0-Signed-off-by: Hyeongseok Oh <hseok82.oh@samsung.com>

---

Draft: https://github.com/Samsung/ONE/pull/12903 
Related issue: #12904 #13039

### CODE REVIEW

**Review by jyoungyun — DISMISSED** [review]
LGTM

File: runtime/onert/core/src/exec/Execution.cc [hunk 1]
```diff
@@ -126,21 +126,22 @@ void Execution::setOutputType(const ir::IOIndex &index, const ir::TypeInfo &type
   _ctx.shape_updated = true;
 }
 
-void Execution::execute()
+void Execution::execute(const ExecutionOptions &options)
```
Comment (ragmani) [inline runtime/onert/core/src/exec/Execution.cc:129]: I have a question.
Is it difficult to set `ExecutionOptions`  only when `Execution` is created? It seems that `ExecutionOptions` is not changed, but is set whenever `execute()` is called.

Comment (hseok-oh) [reply runtime/onert/core/src/exec/Execution.cc:129]: `Execution` is created on `nnfw_prepare()` API, but `ExecutionOptions` can be updated by `nnfw_set_execute_config` APIs. 

To change to set `ExectionOptions` on `Execution` creation, we need to update I/O buffer setting and tensorinfo getter/setter implementation, then we can create `Execution` when `nnfw_run()` is called. It requires many implementation changes.

Comment (hseok-oh) [reply runtime/onert/core/src/exec/Execution.cc:129]: I'll try.

**Review by ragmani — APPROVED** [review]
LGTM

**Review by jyoungyun — APPROVED** [review]
LGTM

**Pull request merged**